Bio/bgzf.py

```diff
@@ -247,19 +247,56 @@
 binary mode, and decode the appropriate fragments yourself.
 """
 
+from __future__ import annotations
+
+import typing as tp
+
 import struct
 import sys
 import zlib
 
 from builtins import open as _open
 
+if tp.TYPE_CHECKING:
+    import _typeshed
+    import io
+    import collections.abc as cx_abc
+    from typing_extensions import TypeAlias, TypeGuard
+
+_BufferT = tp.TypeVar("_BufferT", str, bytes)
+_Self = tp.TypeVar("_Self")
+# fmt: off
+# Writing and updating modes
+_ReadingBinaryMode: TypeAlias = tp.Literal["rb", "br"]
```

Comment:
Wow, esentially an enum - does this work nicely in a compatible code editor for suggesting values? Until now I'd have just annotated this as ``str``.

Reply:
Some language server protocols might work well with `typing.Literal` - I use PyCharm and unfortunately it doesn't seem to do so :(